Source/JavaWrappers.cpp

```diff
@@ -274,6 +281,88 @@ namespace java::lang
     }
 }
 
+namespace java::websocket
+{
+    WebSocketClient::WebSocketClient(std::string url, std::function<void(std::string)> message, std::function<void()> open, std::function<void()> close, std::function<void()> error)
+        : Object{android::global::GetWebSocketClass()}
+        , m_openCallback{std::move(open)}
+        , m_messageCallback{std::move(message)}
+        , m_closeCallback{std::move(close)}
+        , m_errorCallback{std::move(error)}
+    {
+        static JNINativeMethod methods[] = {
+                {"closeCallback", "()V", (void*)OnClose },
+                {"openCallback", "()V", (void*)&WebSocketClient::OnOpen },
+                {"messageCallback", "(Ljava/lang/String;)V", (void*)&WebSocketClient::OnMessage },
+                {"errorCallback", "()V", (void*)&WebSocketClient::OnError },
+        };
+        m_env->RegisterNatives(m_class, methods, 4);
+
+        // initialize socket
+        JObject(m_env->NewObject(m_class, m_env->GetMethodID(m_class, "<init>", "(Ljava/lang/String;)V"), m_env->NewStringUTF(url.c_str())));
+
+        WebSocketClient *p_webSocketClient = this;
+        s_objectMap.insert({JObject(), p_webSocketClient});
+
+        jmethodID connectSocket{m_env->GetMethodID(m_class, "connectBlocking", "()Z")};
```

Comment:
@bghgary not sure if we have a preference for whether we should make the connect call async or sync.

Reply:
Ideally, all functions that might take time should be async to prevent the JavaScript thread from blocking. For some scenarios, there might be rendering happening and it will be noticeable.

Reply:
If you do this as a separate PR, please file an issue to track.